src/mscorlib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -628,6 +628,49 @@ public bool Remove(TKey key)
             return false;
         }
 
+        public bool Remove(TKey key, out TValue value)
+        {
+            if (key == null)
+            {
+                value = default(TValue);
```

Comment:
@jkotas hmmm as I mentioned earlier, in the initial code I did leave out assigning default to the `out` parameter referring to some method but I am not able to find out which. I did the latest change based on `Dictionary.TryGetValue` itself which actually assigned default value on false path (and in fact there is no exception since it is a `Try` method)... please refer [here](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Collections/Generic/Dictionary.cs#L633).

So looked up MSDN for what should be the behavior of an `out` parameter in C# ([link](https://msdn.microsoft.com/en-us/library/t3c3bfhx.aspx)) and it says this...

> Although variables passed as out arguments do not have to be initialized before being passed, the called method is required to assign a value before the method returns.

And then this aligned with the `TryGetValue` behavior mentioned previously and also with COM which mentions that callee is supposed to assign value to out parameter even on E_FAIL, in which case it may be a NULL pointer... hence...